### PR TITLE
Implemented IWallet interface

### PR DESCRIPTION
**Pull request by sirano11**
## Proposed changes

This PR has the following contents.
   - Defined the `IWallet` interface and implemented it to the `KeyringContainer` class.
   - To use `IWallet` interface in `Contract` class, modify return type `Caver@getWallet()`
      - also when signing in `Contract` class, it modified code to `caver.getWallet().sign()`

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/caver-java/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/caver-java)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -72,6 +72,12 @@ configure(subprojects.findAll { it.name != 'integration-test' }) {
         from sourceSets.main.allSource
     }
 
+    task fatJar(type: Jar) {
```
Comment (jimni1222) [inline build.gradle:75]: What is this for?

File: core/src/main/java/com/klaytn/caver/wallet/IWallet.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import com.klaytn.caver.transaction.AbstractFeeDelegatedTransaction;
+import com.klaytn.caver.transaction.AbstractTransaction;
+
+import java.util.List;
+import java.util.function.Function;
+
+public interface IWallet {
+
+    /**
+     * Generates wallet data.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/IWallet.java:12]:
```suggestion
     * Generates data to manage in wallet.
```

File: core/src/main/java/com/klaytn/caver/wallet/IWallet.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+     * @param num The number of data to create.
+     * @return List
+     */
+    List<String> generate(int num);
+
+    /**
+     * Check whether there is a wallet data corresponding to the address passed as a parameter in the wallet.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/IWallet.java:19]:
```suggestion
     * Check whether there is a data corresponding to the address passed as a parameter in the wallet.
```

File: core/src/main/java/com/klaytn/caver/wallet/IWallet.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+     * @param address An address to find wallet data in wallet.
+     * @return boolean
+     */
+    boolean isExisted(String address);
+
+    /**
+     * Deletes the wallet data that associates with the given address from wallet.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/IWallet.java:26]:
```suggestion
     * Deletes the data that associates with the given address from wallet.
```

File: core/src/main/java/com/klaytn/caver/wallet/IWallet.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+     * @param address An address of the wallet data to be deleted in wallet.
+     * @return boolean
+     * @throws Exception
+     */
+    boolean remove(String address) throws Exception;
+
+    /**
+     * Signs the transaction using all keys in the wallet data corresponding to the address.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/IWallet.java:34]:
```suggestion
     * Signs the transaction using all keys in the data corresponding to the address.
```

File: build.gradle (hunk 1 above)
Comment (sirano11) [reply build.gradle:75]: This task make jar contains with all dependency lib. It needs to develop caver-java-ext-kas. 
If you think this task don't need, I will delete this task.

File: core/src/main/java/com/klaytn/caver/wallet/IWallet.java [hunk 6]
```diff
@@ -6,42 +6,45 @@
 import java.util.List;
 import java.util.function.Function;
 
+/**
+ * This interface manages key data.
```
Comment (jimni1222) [inline core/src/main/java/com/klaytn/caver/wallet/IWallet.java:10]:
```suggestion
 * The Interface of wallet that can manage multiple key data.
```

**Review by jimni1222 — DISMISSED** [review]
LGTM except for minor comment :)

**Review by jimni1222 — APPROVED** [review]

**Pull request merged**